requests/src/main/scala/weco/api/requests/responses/LookupPendingRequests.scala

```diff
@@ -20,31 +17,45 @@ trait LookupPendingRequests extends CustomDirectives {
   implicit val ec: ExecutionContext
 
   def lookupRequests(patronNumber: SierraPatronNumber): Route = {
-    val userHolds =
-      for {
-        userHolds <- sierraService.getStacksUserHolds(patronNumber)
-
-        holdsWithCatalogueIds <- Future.sequence(
-          userHolds.right.get.holds.map { hold: StacksHold =>
-            itemLookup
-              .bySourceIdentifier(hold.sourceIdentifier)
-              .map {
-                case Left(elasticError: ElasticsearchError) =>
-                  warn(
-                    s"Unable to look up $hold in Elasticsearch: ${elasticError}"
-                  )
-                  hold
-
-                case Right(item: Item[IdState.Identified]) =>
-                  hold.copy(item = Some(item))
-              }
-          }
+    val itemHolds = for {
+      holdsMap <- sierraService.getHolds(patronNumber)
+
+      itemLookupResults <- Future
+        .sequence(
+          holdsMap.keys
+            .map(srcId => (srcId, itemLookup.bySourceIdentifier(srcId)))
+            .map {
+              case (srcId, itemLookup) =>
```

Comment:
You're reusing the variable name `itemLookup`, which feels like a potential source of confusion.